qt/scientific_interfaces/Inelastic/BayesFitting/Quasi.cpp

```diff
@@ -262,47 +287,32 @@ void Quasi::updateMiniPlot() {
     g_log.warning(ex.what());
   }
 
-  // Update fit plot
-  QString program = m_uiForm.cbProgram->currentText();
-  if (program == "Lorentzians")
-    program = "QL";
-  else
-    program = "QSe";
-
-  QString resName = m_uiForm.dsResolution->getCurrentDataName();
-
-  // Should be either "red", "sqw" or "res"
-  QString resType = resName.right(3);
-
-  // Get the correct workspace name based on the type of resolution file
-  if (program == "QL") {
-    if (resType == "res")
-      program += "r";
-    else
-      program += "d";
-  }
+  auto const fitGroupName = m_outputBaseName + "_Fit";
+  if (!AnalysisDataService::Instance().doesExist(fitGroupName))
+    return;
 
-  QString outWsName = sampleName.left(sampleName.size() - 3) + program + "_Workspace_" + QString::number(m_previewSpec);
-  if (!AnalysisDataService::Instance().doesExist(outWsName.toStdString()))
+  auto const fitGroup = AnalysisDataService::Instance().retrieveWS<WorkspaceGroup>(fitGroupName);
+  if (!fitGroup || fitGroup->getNumberOfEntries() <= m_previewSpec) {
     return;
+  }
 
-  MatrixWorkspace_sptr outputWorkspace =
-      AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>(outWsName.toStdString());
+  auto const outputWorkspace =
+      std::dynamic_pointer_cast<MatrixWorkspace>(fitGroup->getItem(static_cast<std::size_t>(m_previewSpec)));
 
   TextAxis *axis = dynamic_cast<TextAxis *>(outputWorkspace->getAxis(1));
 
   for (std::size_t histIndex = 0; histIndex < outputWorkspace->getNumberHistograms(); histIndex++) {
     QString specName = QString::fromStdString(axis->label(histIndex));
     QColor curveColour;
 
-    if (specName.contains("fit.1"))
+    if (specName.contains("fit 1"))
```

Comment:
I agree with removing dots from naming in this regard, could we add instead underscores? `_`

Reply:
This is also similar to the point above. The BayesQuasi algorithm returned a workspace with labels such as "fit.1", but the BayesQuasi2 algorithm returned a workspace with labels such as "fit 1". So I changed the BayesQuasi algorithm to match the behaviour of the BayesQuasi2 algorithm. Do you think its worth changing to underscores in both algorithms?

Reply:
Ok. I understand, thanks for the explanation. I think is fine as it is.